### PR TITLE
SIM-1748 ADD documentation for new rule preventing JPQL injections.

### PR DESCRIPTION
**Pull request by gregor-balint**

### CODE REVIEW

File: docs/rules/use-parameterized-jpa-query.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+[[toc]]
+
+## Properties
+
+<RuleProperties />
+
+## Description
+
+JPA defines the query language JPQL (Java Persistence Query Language). A JPQL query string may be constructed by concatenating string literals with user defined expressions (e.g. variables, method invocations, user input, etc). Such a query string concatenation in JPQL may cause the same kind of vulnerability to injection attacks as a native SQL query string because the user input may be interpreted as JPQL code.
```
Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:26]: Please use the link to JPQL language reference: 
[JPQL](https://docs.oracle.com/html/E13946_04/ejb3_langref.html)

File: docs/rules/use-parameterized-jpa-query.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+[[toc]]
+
+## Properties
+
+<RuleProperties />
+
+## Description
+
+JPA defines the query language JPQL (Java Persistence Query Language). A JPQL query string may be constructed by concatenating string literals with user defined expressions (e.g. variables, method invocations, user input, etc). Such a query string concatenation in JPQL may cause the same kind of vulnerability to injection attacks as a native SQL query string because the user input may be interpreted as JPQL code.  
+This rule looks for queries of the type [javax.persistence.Query](https://docs.oracle.com/javaee/7/api/javax/persistence/Query.html) which are created by the ```createQuery``` method of [javax.persistence.EntityManager](https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html). Vulnerable components of the JPQL query string concatenation are replaced by parametrizing, using the ```setParameter``` method of ```javax.persistence.Query```.
```
Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:27]: Please link the createQuery and setParameter.

Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:27]: In my opinion, this sentence is not clear: 

> Vulnerable components of the JPQL query string concatenation are replaced by parametrizing, using the setParameter method of Query.
Would you consider this instead: 

> The vulnerable concats of the JPQL query strings are parameterized, thus they can only be considered as data and not as code.

Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:27]: Please use single backticks for inline code style.

File: docs/rules/use-parameterized-jpa-query.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+## Benefits
+
+Prevents SQL injections when using the Java Persistence API (JPA)
```
Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:31]: The dot at the end of the sentence is missing.

File: docs/rules/use-parameterized-jpa-query.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ruleId: UseParameterizedJPAQuery
+since: 3.18.0
+minJavaVersion: 1.1
+remediationCost: 10
+links:
+    - displayName: "How To Fix SQL Injection: JPA"
+      url: "https://software-security.sans.org/developer-how-to/fix-sql-injection-in-java-persistence-api-jpa"
+    
+description:
+    Rule description.
```
Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:12]: Please write the description here.

Comment (gregor-balint) [reply docs/rules/use-parameterized-jpa-query.md:12]: I put a short description here.

File: docs/rules/use-parameterized-jpa-query.md (hunk 2 above)
Comment (gregor-balint) [reply docs/rules/use-parameterized-jpa-query.md:27]: I added all missing links to JPQL (Java Persistence Query Language), createQuery and setParameter.

Comment (gregor-balint) [reply docs/rules/use-parameterized-jpa-query.md:27]: I made the change similar to Your recommendation...

File: docs/rules/use-parameterized-jpa-query.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+JPA defines the query language [JPQL (Java Persistence Query Language)](https://docs.oracle.com/html/E13946_04/ejb3_langref.html). A JPQL query string may be constructed by concatenating string literals with user defined expressions (e.g. variables, method invocations, user input, etc). Such a query string concatenation in JPQL may cause the same kind of vulnerability to injection attacks as a native SQL query string because the user input may be interpreted as JPQL code.  
+This rule looks for queries of the type [javax.persistence.Query](https://docs.oracle.com/javaee/7/api/javax/persistence/Query.html) which are created by the [createQuery](https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#createQuery-java.lang.String-) method of [javax.persistence.EntityManager](https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html). The vulnerable concats of the JPQL query strings are parameterized, thus they can only be considered as data and not as code. For this purpose, the method [setParameter](https://docs.oracle.com/javaee/7/api/javax/persistence/Query.html#setParameter-int-java.lang.Object-) of the class `javax.persistence.Query` is used.
+
```
Comment (arditymeri) [inline docs/rules/use-parameterized-jpa-query.md:28]: IMO, the last sentence 'Fort this purpose...' is not necessary.

**Pull request merged**